enterprise/models.py

```diff
@@ -2,42 +2,235 @@
+                                    parts of the system (SSO, ecommerce, analytics etc.)
+        logo (ImageField):  Enterprise Customer logo
+        data_sharing_policy (CharField): stores data sharing policy: either no data sharing,
+                                         requested data sharing (learners are allowed to opt-in) or required data
+                                         sharing (learners must share their data)
+        active (BooleanField): used to mark inactive Enterprise Customers - implements "soft delete" pattern
+        members (ManyToMany[User]): list of Enterprise Customer members.
     """
 
-    # TODO: add field definitions
+    NO_DATA_SHARING = "no_sharing"
+    REQUEST_DATA_SHARING = "request"
+    REQUIRE_DATA_SHARING = "require"
+
+    class Meta(object):
+        """
+        Meta class for EnterpriseCustomer model.
+        """
+
+        app_label = "enterprise"
+        verbose_name = "Enterprise Customer"
+        verbose_name_plural = "Enterprise Customers"
+
+    objects = EnterpriseCustomerManager()
+    all_objects = models.Manager()
+
+    name = models.CharField(max_length=500, blank=False, null=False, help_text=_("Enterprise Customer name."))
+    short_code = AutoSlugField(
+        max_length=120, blank=True, null=False, editable=True,
+        populate_from="name", unique=True,
+        help_text=_("Short code will be used to reference this Enterprise Customer in other parts of the system,"
+                    "i.e. SSO, ecommerce app, analytics etc. It is allowed to override it at creation, but it should "
+                    "never be edited on existing records, as it might break integration with other apps.")
+    )
+    logo = models.ImageField(
+        upload_to="enterprise_logos",
+        help_text=_("Please add only .PNG files for logo images."),
+        null=True, blank=True, max_length=255
+    )
+    data_sharing_policy = models.CharField(
+        blank=False, null=False, help_text=_("Data sharing policy"), max_length=20,
+        choices=(
+            (NO_DATA_SHARING, _("No data sharing requested")),
+            (REQUEST_DATA_SHARING, _("Request data sharing")),
+            (REQUIRE_DATA_SHARING, _("Require data sharing")),
+        )
+    )
+    active = models.BooleanField(default=True)
 
     def __str__(self):
         """
-        Get a string representation of this model instance.
+        Return human-readable string representation.
+        """
+        return "<EnterpriseCustomer {ID}: {name}>".format(ID=self.id, name=self.name)
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
         """
-        # TODO: return a string appropriate for the data fields
-        return '<EnterpriseCustomer, ID: {}>'.format(self.id)
+        return self.__str__()
 
 
 @python_2_unicode_compatible
 class EnterpriseCustomerHistory(TimeStampedModel):
```

Comment:
I see no benefit to recreating the wheel here. Use [`django-simple-history`](https://django-simple-history.readthedocs.io/en/latest/).